### ISSUE TITLE
Rule: Warn on late definitions of variables/functions

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#3]
This rule warns when a variable or function is used before it is declared, such as:

``` js
doIt();

function doIt() {
}
```

Rule name should be `no-latedef`
